analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.IsNull.cs

```diff
@@ -89,8 +87,31 @@ public void IsNull_Coalesce_UnknownToUnknown()
 Tag(""UnknownToUnknown"", unknownToUnknown);";
             var validator = SETestContext.CreateCS(code, ", object arg1, object arg2").Validator;
             validator.ValidateContainsOperation(OperationKind.IsNull);
-            validator.TagValues("UnknownToUnknown").Should().HaveCount(1)
-                .And.ContainSingle(x => x == null);
+            validator.TagValues("UnknownToUnknown").Should().SatisfyRespectively(
+                x => x.Should().HaveNoConstraints(),
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
```

Comment:
Here also: It would be interesting to see the constraints of the single symbols.

Reply:
I also found this case while working on #7039 (fd523d6). The "NoConstraints" case causes trouble in S3900. I left a comment in https://github.com/SonarSource/sonar-dotnet/pull/7039#discussion_r1158418491 on how to troubleshoot this for S3900.